### DISCUSSION TITLE
Damaged boot loader on PowerSpec?

### DISCUSSION THREAD

**Discussion by Mrnt** [moonglow/FlashForge_Marlin#98] [Q&A]
Initially I was able to flash the firmware using the ff_flash_firmware_tool.py, but at some point during my attempts it seems the boot loader has been damaged slightly. 
These are the symptoms:
1) At boot the start screen displays a check id fail message:
![IMG_5743](https://user-images.githubusercontent.com/1972570/157750024-ed3d12bf-7da7-4e1b-bb3d-8c37fb0187aa.jpeg)
but it will continue to start up.
2) I can use the ff_flash_firmware_tool.py to copy firmware to the internal SD card, but the machine will not reboot automatically after copying the firmware. If I reboot the machine manually I see a screen indicating that the firmware is being written from the SD card to the machine:
![IMG_5744](https://user-images.githubusercontent.com/1972570/157750049-2c48b611-858a-4147-ab8e-42bc470e4be1.jpeg)
Then it hangs and I have to turn it off, remove the internal SD card and erase the firmware file (in this case power.bin) and then if I put the SD card back it will boot normally (apart from the error message in (1) above) with the new firmware.

I have tried installing the powerspec_v2.2.20150612.bin firmware file from PowerPrint, but no difference to the symptoms.

Questions:
Does it sound like the bootloader is damaged? 
If so, do you have any ideas on how I would reinstall it?

**Comment by moonglow** [comment]
First of all i recommend you to backup full flash firmware ( [LINK](https://github.com/moonglow/FlashForge_Marlin/wiki/Backup-printer-firmware) ) and you can do anything you want. But i will add such function from Marlin menu to backup firmware bootloader without JTAG.

> Does it sound like the bootloader is damaged?

Nope. Their bootloader has a motherboard ID check mechanism ( based on MCU UID ) that value saved just after bootloader location and it is not really good solution because it can be accidentally damaged by bootloader itself by standard update procedure to any firmware ( official or not official )

> If so, do you have any ideas on how I would reinstall it?

I will try to add to my Marlin firmware such option or you can dump your bootloader via JTAG and i will patch it to skip such check.
